src/modules/ods/ods.service.ts

```diff
@@ -0,0 +1,90 @@
+import { DATABASE } from '@ukef/constants';
+import { DataSource } from 'typeorm';
+import { PinoLogger } from 'nestjs-pino';
+import { ODS_ENTITIES, OdsEntity, odsStoredProcedureInput, odsStoredProcedureQueryParams } from './dto/ods-payloads.dto';
+
+@Injectable()
+export class OdsService {
+  constructor(
+    @InjectDataSource(DATABASE.ODS)
+    private readonly odsDataSource: DataSource,
+    private readonly logger: PinoLogger,
+  ) {}
+
+  async findCustomer(partyUrn: string): Promise<GetOdsCustomerResponse> {
+    try {
+      const spInput = this.createOdsStoredProcedureInput(ODS_ENTITIES.CUSTOMER, { customer_party_unique_reference_number: partyUrn });
+
+      const storedProcedureResult = await this.callOdsStoredProcedure(spInput);
+
+      const storedProcedureJson = JSON.parse(storedProcedureResult[0]?.output_body);
+
+      if (storedProcedureJson == undefined || storedProcedureJson?.status != 'SUCCESS') {
+        this.logger.error('Error from ODS stored procedure, output: %o', storedProcedureResult);
+        throw new InternalServerErrorException('Error trying to find a customer');
+      }
+
+      if (storedProcedureJson?.total_result_count == 0) {
+        throw new NotFoundException('No matching customer found');
+      }
+
+      return {
+        customerUrn: storedProcedureJson.results[0]?.customer_party_unique_reference_number,
+        customerName: storedProcedureJson.results[0]?.customer_name,
+      };
+    } catch (err) {
+      if (err instanceof NotFoundException) {
+        this.logger.warn(err);
+        throw err;
+      } else {
+        this.logger.error(err);
+        throw new InternalServerErrorException('Error trying to find a customer');
+      }
+    }
+  }
+
+  /**
+   * Creates the input parameter for the stored procedure
+   * @param {OdsEntity} entityToQuery The entity you want to query in ODS
+   * @param {odsStoredProcedureQueryParams} queryParameters The query parameters and filters to apply to the query
+   *
+   * @returns {odsStoredProcedureInput} The ODS stored procedure input in object format
+   */
+  createOdsStoredProcedureInput(entityToQuery: OdsEntity, queryParameters: odsStoredProcedureQueryParams): odsStoredProcedureInput {
+    return {
+      query_method: 'get',
+      query_object: entityToQuery,
+      query_page_size: 1,
+      query_page_index: 1,
+      query_parameters: queryParameters,
+    };
+  }
+
+  /**
+   * Calls the ODS stored procedure with the input provided and returns the output of it
+   * @param {odsStoredProcedureInput} storedProcedureInput The input parameter of the stored procedure
+   *
+   * @returns {Promise<any>} The result of the stored procedure
+   */
+  async callOdsStoredProcedure(storedProcedureInput: odsStoredProcedureInput): Promise<any> {
+    const queryRunner = this.odsDataSource.createQueryRunner();
+    try {
+      // Use the query runner to call a stored procedure
+      const result = await queryRunner.query(
```

Comment:
Unfortuantely typeorm doesn't support any calling of stored procedures outside of just a SQL query so I've written a parameterised SQL query to stop SQL injection but has to be hard coded like this